src/main/scala/com.snowplowanalytics/iglu.schemaddl/migrations/Migration.scala

```diff
@@ -36,88 +37,88 @@ case class Migration(vendor: String, name: String, from: SchemaVer.Full, to: Sch
    * This method requires all intermediate schemas because we need to keep an order of properties
    */
-  def buildMigration(source: OrderedSchemas): Migration = {
+  def buildMigration(source: SchemaListSegment): Migration = {
     val base = source.schemas.head.self.schemaKey
     val diff = SchemaDiff.build(source)
     Migration(base.vendor, base.name, base.version, source.schemas.last.self.schemaKey.version, diff)
   }
 
-  // NEL has minimum 2 elements
-  def buildMatrix[A](as: List[A]) = {
-    val ordered = as.zipWithIndex
-    for {
-      (from, fromIdx) <- ordered
-      (to,   toIdx)   <- ordered
-      cell <- NonEmptyList.fromList(ordered.filter { case (_, i) => i >= fromIdx && i <= toIdx }) match {
-        case None => Nil
-        case Some(NonEmptyList(_, Nil)) => Nil
-        case Some(nel) => List((from, to, nel.map(_._1)))
-      }
-    } yield cell
-  }
-
-  def buildMigrationMatrix(schemas: NonEmptyList[IgluSchema]): NonEmptyList[OrderedSchemas] =
-    schemas
-      .groupByNem { case SelfDescribingSchema(SchemaMap(SchemaKey(v, n, _, SchemaVer.Full(model, _, _))), _) => (v, n, model) }
-      .toNel
-      .flatMap { case (_, s) =>
-        val matrix = buildMatrix(s.toList).map { case (_, _, ss) => OrderedSchemas(ss) }
-        NonEmptyList.fromListUnsafe(matrix)
-      }
-
   /**
    * Build [[MigrationMap]], a map of source Schema to it's migrations,
    * where all source Schemas belong to a single model-revision Schema criterion
    *
-   * @param schemas source Schemas belong to a single model-revision criterion
-   * @return migration map of each Schema to list of all available migrations
+   * @param schemaListFulls List of SchemaListFull to create migrations
+   * @return map of each Schema to list of all available migrations
    */
-  def buildMigrationMap(schemas: List[IgluSchema]): MigrationMap =
-    buildMigrationMatrix(NonEmptyList.fromListUnsafe(schemas))
-      .map(source => (source.schemas.head.self, buildMigration(source)))
+  def buildMigrationMap(schemaListFulls: NonEmptyList[SchemaListFull]): MigrationMap = {
+    // groupBy of NonEmptyList requires cats.Order of SchemaMap
+    // however there is no dependency like this in groupBy of Scala List.
+    // Therefore, it is converted to Scala List initially and reconverted
+    // to NonEmptyList afterward.
+    buildMigrationMatrix(schemaListFulls)
+      .toList.map(source => (source.schemas.head.self, buildMigration(source)))
       .groupBy(_._1)
-      .mapValues(_.map(_._2))
+      .mapValues(m => NonEmptyList.fromListUnsafe(m.map(_._2)))
+  }
 
+  /**
+    * Build a map of source Schema to its OrderedSubSchemas, where all source Schemas
+    * are last version of their model group
+    * @param orderedSchemasList source Schemas
+    * @return map of last version of Schema model group to its OrderedSubSchemas
+    */
+  def buildOrderedSubSchemasMap(orderedSchemasList: NonEmptyList[SchemaList]): Map[SchemaMap, OrderedSubSchemas] =
```

Comment:
Where is this function used?

Reply:
It is used in the IgluCtl generate command.

Reply:
Ok. Didn't notice it. Do you think we can/should move it to igluctl or make it private or/and combine with other function(s)? My problem is just that its purpose is fairly cryptic yet simple in implementation. I would like to have only functions that are core/basic to actual Schema DDL goals.

Reply:
Yep, I think you are right. It would be better to move it to IgluCtl. I will move it 👍

Reply:
That's totally up to you though, not insisting on anything.